src/faucet/utils.rs

```diff
@@ -122,6 +123,11 @@ pub fn format_balance(balance: &TokenAmount, unit: &str) -> String {
     )
 }
 
+/// Constructs a URL to lookup the faucet history for a given address.
+pub fn format_address_url(base_url: &Url, address: &str) -> Result<Url, ParseError> {
+    base_url.join(&format!("address/{}", address))
+}
```

Comment:
Any reason why you are not using the `anyhow::Result`?

Reply:
updated and added test now

Reply:
I also introduced `SearchPath` which now has only one variant but can be extended for other search types like transaction, tipset, etc